specutils/analysis/width.py

```diff
@@ -0,0 +1,190 @@
+
+    Returns
+    -------
+    ew : `~astropy.units.Quantity`
+        Equivalent width calculation.
+    """
+    return _computation_wrapper(_compute_equivalent_width, spectrum, region)
+
+
+def _compute_full_width_half_max(spectrum, region=None):
+    """
+    Calculate the full width of the spectrum at half max.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the FWHM value.
+
+    Returns
+    -------
+    full_width_half_max : `~astropy.units.Quantity` or list (based on region input)
+        Approximate full width of the signal at half max
+
+    Notes
+    -----
+    This is a helper function for the above `full_width_half_max()` method.
+
+    """
+
+    if region is not None:
+        calc_spectrum = region.extract(spectrum)
+    else:
+        calc_spectrum = spectrum
+
+    flux = calc_spectrum.flux
+    frequencies = calc_spectrum.frequency
+
+    dx = frequencies - np.mean(frequencies)
+    fwhm = 2 * np.sqrt(np.sum((dx * dx) * flux, axis=-1) / np.sum(flux, axis=-1))
+
+    return fwhm
+
+
+def _compute_sigma_full_width(spectrum, region=None):
+    """
+    Calculate the full width of the spectrum based on an approximate value of
+    sigma.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the gaussian sigma width.
+
+    Returns
+    -------
+    full_width : `~astropy.units.Quantity` or list (based on region input)
+        Approximate full width of the signal
+
+    Notes
+    -----
+    This is a helper function for the above `sigma_full_width()` method.
+
+    """
+
+    fwhm = _compute_full_width_half_max(spectrum, region)
+    sigma = fwhm * gaussian_fwhm_to_sigma
+
+    return sigma * 2
+
+
+def _compute_equivalent_width(spectrum, region=None):
+    """
+    Compute a naive equivalent width measures on the spectrum object.
+
+    Parameters
+    ----------
+    spectrum : Spectrum1D
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the gaussian sigma width.
+
+    Returns
+    -------
+    ew : `~astropy.units.Quantity`
+        Equivalent width calculation.
+
+    TODO:  what frame of reference do you want the spectral_axis to be in ???
+    """
+
+    if region is not None:
+        calc_spectrum = region.extract(spectrum)
+    else:
+        calc_spectrum = spectrum
+
+    flux = calc_spectrum.flux
+    wavelength = calc_spectrum.wavelength
+
+    # Continuum is always assumed to be 1.0
+    avg_cont = np.median(flux)
```

Comment:
I would say instead of assuming the `median`, have it be a user-supplied value, which should default to ``1.0``.  Might leave in the `median` option by being a special case if the user gives the string `median`, though.